whitepapers/0011_accountant.md

```diff
@@ -0,0 +1,190 @@
+
+Example 4: A user sends wrapped SOL from ethereum to polygon.  Even though SOL is not native to either chain, the transaction is still recorded the same way: the `Ethereum/SOL` mint account is debited and the `Polygon/SOL` mint account is credited.  The total liability of the token bridge has not changed and instead we have just moved some liability from ethereum to polygon.
+
+#### Creating accounts
+
+Since users can send any arbitrary token over the token bridge, the accountant should automatically create accounts for any new token.  To determine whether the newly created account should be a custody account or a mint account, these steps are followed:
+
+- If `emitter_chain == token_chain` and a custody account doesn't already exist, create one.
+- If `emitter_chain != token_chain` and a wrapped account doesn't already exist, reject the transaction.  The wrapped account should have been created when the tokens were originally transferred to `emitter_chain`.
+- If `token_chain != recipient_chain` and a wrapped account doesn't already exist, create one.
+- If `token_chain == recipient_chain` and a custody account doesn't already exist, reject the transaction.  The custody account should have been created when the tokens were first transferred from `recipient_chain`.
+
+### Bootstrapping / Backfills
+
+In order for the accountant to have accurate values for the balances of the various accounts it needs to replay all transactions since the initial launch of the token bridge.  Similarly, it may be necessary to backfill missing transfers if the accountant is disabled for any period of time.
+
+To handle both issues, the accountant has the ability to directly process tokenbridge transfer VAAs.  As long as the VAAs are signed by a quorum of guardians from the latest guardian set, the accountant uses them to update its on-chain state of token balances.
+
+Submitting VAAs does not disable any core checks: a VAA causing the balance of an account to become negative will still be rejected even if it has a quorum of signatures.  This can be used to catch problematic transfers after the fact.
+
+### Handling Rejections
+
+Legitimate transactions should never be rejected so a rejection implies either a bug in the accountant or an actual attack on the bridge.  In both cases manual intervention is required.
+
+#### Alerting
+
+Whenever a guardian observes an error (either via an observation response or via an error event), it will increase a prometheus error counter.  Guardians will be encouraged to set up alerts whenever these metrics change so that they can be notified when an error occurs.
+
+## Deployment Plan
```

Comment:
Not sure if this section is really relevant for describing the feature.  Especially now that it's already live in enforcing mode.

Reply:
It seems useful if not only to inform users how this was deployed and to show that the wormhole protocol isn't running a 🤡 show. It is also pretty much exactly what we did if I understand things correctly.

Happy to remove it if you have strong opinions. What do you think @jynnantonix?

Reply:
I don't feel that strongly either way.  It's just that the rest of this document describes the feature while only this section describes a plan.  

We can keep it in if people will find it useful.

Reply:
My vote is to leave it, if not only because it follows the template and structure of the other papers. It isn't terribly useful when it is merged, but we don't update older papers to remove the deployment steps post-go-live either.

Happy to change anything if you disagree.